### PR TITLE
upgrade to django 1.11 [ch396]

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

**Comment by dpoirier** [comment]
@domdinicola can you fix the tests?

**Comment by domdinicola** [comment]
@dpoirier if you are not busy can you check the tests?
I fixed all of them, except those related to queries.. looking at them it looks like like the relationship between django and tenant schema changed.

**Comment by dpoirier** [comment]
@domdinicola maybe you could run at least one of those failing tests on Django 1.10 and print out the captured queries to see what's different from running on Django 1.11.

**Comment by ntrncic** [comment]
@dpoirier could you please take over this branch and fix the tests that are failing? thank you :)

File: EquiTrack/partners/tests/test_api_prp.py [hunk 1]
```diff
@@ -105,7 +106,17 @@ def test_prp_api_modified_queries(self):
...
+        Make sure number of queries is as expected
+
+        Queries with Django 1.10 = 21
+        Queries with Django 1.11 = 22
+
+        There's an extra "SET search_path = test,public" with Django 1.11 - why?  The tests in 'environment'
+        app didn't add any extra SET search_path with Django 1.11, once I added TENANT_LIMIT_SET_CALLS = True.
+        """
+
+        EXPECTED_QUERIES = 21 if django.VERSION < (1, 11) else 22
```
Comment (domdinicola) [inline EquiTrack/partners/tests/test_api_prp.py]: should we just set this to 22, the comment should be enough

Comment (dpoirier) [reply EquiTrack/partners/tests/test_api_prp.py]: assertNumQueries is an exact assertion, not a "no more than this many" assertion, so we need to assert the exact number of queries we're going to make for each version.

Comment (domdinicola) [reply EquiTrack/partners/tests/test_api_prp.py]: if we're migrating to dj1.11 I would expect this to be always 22
we are not using tox to run test on dj1.10 and dj1.11

Comment (dpoirier) [reply EquiTrack/partners/tests/test_api_prp.py]: I see, I misunderstood the comment.

Comment (domdinicola) [reply EquiTrack/partners/tests/test_api_prp.py]: I could have explain it better, thank you!

File: EquiTrack/utils/writable_serializers/tests/models.py [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
+    content_type = models.ForeignKey(ContentType, related_name='generic_child')
     obj = GenericForeignKey()
 
     field = models.IntegerField()
 
 
 class CodedGenericChild(models.Model):
     object_id = models.IntegerField()
-    content_type = models.ForeignKey(ContentType)
+    content_type = models.ForeignKey(ContentType, related_name='coded_generic_chield')
```
Comment (domdinicola) [inline EquiTrack/utils/writable_serializers/tests/models.py]: @dpoirier can you please fix the related name (my bad)

Comment (dpoirier) [reply EquiTrack/utils/writable_serializers/tests/models.py]: @domdinicola updated

**Comment by domdinicola** [comment]
@dpoirier  can you check why is it failing now?

**Comment by dpoirier** [comment]
It looks like django-tenant-schemas has some unresolved problems with Django 1.11.  We might need to wait for a new release of django-tenant-schemas before we can upgrade Django.

e.g. https://github.com/bernardopires/django-tenant-schemas/pull/524, https://github.com/bernardopires/django-tenant-schemas/pull/519

File: EquiTrack/utils/writable_serializers/tests/models.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ class GenericChild(models.Model):
 
 class CodedGenericChild(models.Model):
     object_id = models.IntegerField()
-    content_type = models.ForeignKey(ContentType, related_name='coded_generic_chield')
+    content_type = models.ForeignKey(ContentType, related_name='coded_generic_child')
```
Comment (domdinicola) [inline EquiTrack/utils/writable_serializers/tests/models.py]: thank you!

**Comment by robertavram** [comment]
@dpoirier I'm putting this on hold until someone confirms that tenant schemas is safe..

**Comment by robertavram** [comment]
are we sure tenant-schemas issue affects us?

**Comment by dpoirier** [comment]
@robertavram initial migrations are failing in CI (https://circleci.com/gh/unicef/etools/7821) due to confusion about which tables already exist (apparently), which seems like it could be caused by https://github.com/bernardopires/django-tenant-schemas/pull/519 which is to update tenant-schemas for changes in Postgresql introspection.  But the details of the pull request are over my head. I guess we could try applying the pull request to a local copy and see if it fixes our CI problem?

**Comment by domdinicola** [comment]
@robertavram sure! not problem with that

**Comment by domdinicola** [comment]
[ch396]

**Pull request closed without merging**